### PR TITLE
Convert the event name to lower case

### PR DESCRIPTION
**Pull request by hvardhan-unth**
HubSpot requires that their custom behavioral objects to be prefixed with pe_<account_id>_{lowercased event_name}, but currently it's not possible to do this for events that are in camelCase, so the events are not successfully being received by HubSpot as the customer’s configured custom behavioral events. 
So, in this PR the event name is changed into lower case.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)

### CODE REVIEW

File: packages/cli/src/commands/generate/action.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ export default class GenerateAction extends Command {
         message: 'Which integration (directory)?',
         choices: integrationDirs.map((integrationPath) => {
           const [name] = integrationPath.split(path.sep).reverse()
-          const value = isBrowserDestination ? path.join(integrationPath, 'src') : integrationPath;
+          const value = isBrowserDestination ? path.join(integrationPath, 'src') : integrationPath
```
Comment (varadarajan-tw) [inline packages/cli/src/commands/generate/action.ts:74]: You might want to revert this change.

File: packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts [hunk 2]
```diff
@@ -68,6 +68,8 @@ const action: ActionDefinition<Settings, Payload> = {
     }
   },
   perform: (request, { payload, settings }) => {
+    payload.eventName = payload.eventName.replace(/[\s.]+/g, '_').toLocaleLowerCase()
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts:71]: Would it affect existing mappings that customers have setup? Could you do some regression testing in stage?
Also could you add a unit test?

Comment (hvardhan-unth) [reply packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts]: Added unit test.

Screenshot of testing on stage:-

Before Deploying changes:
Screenshot 2023-07-18 at 5 06 13 PM


After Deploying the changes:
Screenshot 2023-07-21 at 2 00 51 PM


The changes doesn't affect the existing mappings of customer.

File: packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts [hunk 3]
```diff
@@ -68,6 +68,9 @@ const action: ActionDefinition<Settings, Payload> = {
     }
   },
   perform: (request, { payload, settings }) => {
+    //Transform the eventName
+    payload.eventName = transformEventName(payload.eventName)
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts:72]: Mutating payload is not a good idea. Can we instead assign it to a variable instead and reference it?

Comment (varadarajan-tw) [reply packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts:72]: If possible, can we release this behind a feature flag? Just want to be safe.

File: packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts [hunk 4]
```diff
@@ -67,9 +67,15 @@ const action: ActionDefinition<Settings, Payload> = {
       defaultObjectUI: 'keyvalue:only'
     }
   },
-  perform: (request, { payload, settings }) => {
+  perform: (request, { payload, settings, features }) => {
+    let transformedEventName
+    if (features && features['actions-hubspot-event-name']) {
+      //Transform the eventName
+      transformedEventName = transformEventName(payload.eventName)
+    }
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts:75]: nit
```suggestion
    const shouldTransformEventName =  features && features['actions-hubspot-event-name']
    const eventName = shouldTransformEventName ? transformEventName(payload.eventName) : payload.eventName
```

File: packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts [hunk 5]
```diff
@@ -67,9 +67,15 @@ const action: ActionDefinition<Settings, Payload> = {
...
+
     const event: CustomBehavioralEvent = {
-      eventName: payload.eventName,
+      eventName: transformedEventName ? transformedEventName : payload.eventName,
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/hubspot/sendCustomBehavioralEvent/index.ts:78]: Shouldn't the transformedEventName be used in the validations below instead of payload.eventName?

**Review by varadarajan-tw — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
Hi @hvardhan-unth this PR has been deployed.
